app/employee-app/backend/modules/types/types.bal

```diff
@@ -0,0 +1,395 @@
+    BusinessUnit[] businessUnits;
+    # Career paths of the organization
+    CareerFunction[] careerPaths;
+    # Companies of the organization
+    Company[] companies;
+    # Employment types
+    string[] employmentTypes;
+|};
+
+# Recruit type.
+public type Recruit record {
+    # Id of the recruit
+    int recruitId;
+    # First name of the employee
+    string firstName;
+    # Last name of the employee
+    string lastName;
+    # employee's gender
+    string gender;
+    # Personal email of the employee
+    string personalEmail;
+    # WSO2 email of the employee if ex employee
+    string? wso2Email;
+    # Contact number of the employee
+    string contactNumber;
+    # Employment type of the employee
+    string employmentType;
+    # Career function id of the employee
+    int careerFunctionId;
+    # Career function of the employee
+    string careerFunction;
+    # Job band of joining
+    int jobBand;
+    # Designation of joining
+    string designation;
+    # Company id of joining
+    int companyId;
+    # Company of joining
+    string company;
+    # Location of the company
+    string companyLocation;
+    # Office id of joining
+    int officeId;
+    # Office of joining
+    string office;
+    # Office location of joining
+    string officeLocation;
+    # Work location of the employee
+    string workLocation;
+    # Business unit id of joining
+    int businessUnitId;
+    # Business unit of joining
+    string businessUnit;
+    # Department id of joining
+    int departmentId;
+    # Department of joining
+    string department;
+    # Team id of joining
+    int teamId;
+    # Team of joining
+    string team;
+    # Sub team id of joining if exists
+    int? subTeamId;
+    # Sub team of joining if exists
+    string? subTeam;
+    # Manager email of the employee
+    string managerEmail;
+    # Reports to email of the employee
+    string reportsTo;
+    # Date of joining
+    string dateOfJoin;
+    # End date of the probation period
+    string? probationEndDate;
+    # End date of the agreement
+    string? agreementEndDate;
+    # Compensation details of the employee
+    Compensation[] compensation;
+    # Offer related documents
+    Document[]? offerDocuments;
+    # Additional comments
+    string? additionalComments;
+    # Status of the employee
+    string status;
+    # Created by
+    string createdBy;
+    # Created on
+    string createdOn;
+    # Updated by
+    string updatedBy;
+    # Updated on
+    string updatedOn;
+};
+
+# SubTeam.
+public type SubTeam record {
+    # Id of the sub team
+    int id;
+    # Name of the sub team
+    string subTeam;
+};
+
+# Team.
+public type Team record {
+    # Id of the team
+    int id;
+    # Name of the team
+    string team;
+    # Sub teams of the team
+    SubTeam[]? subTeams;
+};
+
+# Update employee payload.
+public type UpdateRecruitPayload record {|
+    # First name of the employee
+    string? firstName = ();
```

Comment:
Are these really nilable?

If these are optional instead, we should model accordingly.

Reply:
This is used in "patch recruits/[int recruited]". Which is the update endpoint. Patches allow us to update any attribute. Shouldn't we use a record with optional parameters to cater this requirement?

Reply:
This record has removed.